hack/e2e.go

```diff
@@ -19,63 +19,63 @@ package main
 		"By default, verify that the cluster has at least two nodes."+
 		"You can explicitly set to false if you're, e.g., testing single-node clusters "+
 		"for which the node count is supposed to be one.")
-
-	ctlCmd = flag.String("ctl", "", "If nonempty, pass this as an argument, and call kubectl. Implies -v. (-test, -cfg, -ctl are mutually exclusive)")
+	checkVersionSkew = flag.Bool("check_version_skew", true, ""+
+		"By default, verify that client and server have exact version match. "+
+		"You can explicitly set to false if you're, e.g., testing client changes "+
+		"for which the server version doesn't make a difference.")
+	checkLeakedResources = flag.Bool("check_leaked_resources", false, "Ensure project ends with the same resources")
+	ctlCmd               = flag.String("ctl", "", "If nonempty, pass this as an argument, and call kubectl. Implies -v. (-test, -cfg, -ctl are mutually exclusive)")
+	down                 = flag.Bool("down", false, "If true, tear down the cluster before exiting.")
+	dump                 = flag.String("dump", "", "If set, dump cluster logs to this location")
+	kubemark             = flag.Bool("kubemark", false, "If true, run kubemark tests.")
+	isup                 = flag.Bool("isup", false, "Check to see if the e2e cluster is up, then exit.")
+	push                 = flag.Bool("push", false, "If true, push to e2e cluster. Has no effect if -up is true.")
+	pushup               = flag.Bool("pushup", false, "If true, push to e2e cluster if it's up, otherwise start the e2e cluster.")
+	skewTests            = flag.Bool("skew", false, "If true, run tests in another version at ../kubernetes/hack/e2e.go")
+	testArgs             = flag.String("test_args", "", "Space-separated list of arguments to pass to Ginkgo test runner.")
+	test                 = flag.Bool("test", false, "Run Ginkgo tests.")
+	up                   = flag.Bool("up", false, "If true, start the the e2e cluster. If cluster is already up, recreate it.")
+	upgradeArgs          = flag.String("upgrade_args", "", "If set, run upgrade tests before other tests")
+	verbose              = flag.Bool("v", false, "If true, print all command output.")
 )
 
 const (
 	minNodeCount = 2
 )
 
-func absOrDie(path string) string {
-	out, err := filepath.Abs(path)
-	if err != nil {
-		panic(err)
-	}
-	return out
-}
-
-type TestResult struct {
-	Pass int
-	Fail int
-}
-
-type ResultsByTest map[string]TestResult
-
 func main() {
 	log.SetFlags(log.LstdFlags | log.Lshortfile)
 	flag.Parse()
 
-	os.Setenv("KUBECTL", *root+`/cluster/kubectl.sh`+kubectlArgs())
+	cwd, err := os.Getwd()
+	if err != nil {
+		log.Fatalf("Could not get pwd: %v", err)
+	}
+	acwd, err := filepath.Abs(cwd)
+	if err != nil {
+		log.Fatalf("Failed to convert to an absolute path: %v", err)
+	}
+	if !strings.Contains(filepath.Base(acwd), "kubernetes") {
+		// TODO(fejta): cd up into  the kubernetes directory
+		log.Fatalf("Must run from kubernetes directory: %v", cwd)
```

Comment:
Could we maybe check for some well known kubernetes release artifacts instead?  Otherwise somewhere we ought to doc that "you must run tests from a directory that has the word kubernetes somewhere in its ancestry"